packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx

```diff
@@ -34,30 +41,40 @@ export const ColorPicker = ({ color, onSelectColor, label }: IColorPickerProps):
    * @returns
    */
   const isValidHexCode = (hexCode: string) => {
-    const hexRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
+    const hexRegex = /^#([A-Fa-f0-9]{6})$/;
     return hexRegex.test(hexCode);
   };
 
-  const handleClick = () => {
-    setShowPicker(!showPicker);
-  };
-
-  const handleColorChange = (color) => {
-    setNewColor(color.hex);
-    onSelectColor(color.hex);
-  };
+  const handleOutsideClick = useCallback((event: MouseEvent) => {
+    const target = event.target as HTMLElement;
+    const pickerContainer = document.getElementById('circle-picker');
+    if (pickerContainer && !pickerContainer.contains(target)) {
+      setShowPicker(false);
+    }
+  }, []);
 
-  const handleHexCodeChange = useCallback((event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
-    setNewColor(event.detail.value);
-    if (isValidHexCode(event.detail.value)) {
-      setHexCodeError(''); // Clear any existing error message
-      onSelectColor(event.detail.value);
+  useEffect(() => {
+    if (showPicker) {
+      document.addEventListener('click', handleOutsideClick);
     } else {
-      setHexCodeError(
-        intl.formatMessage({ defaultMessage: 'Invalid hex code', description: 'hex validations messages' }),
-      ); // Set the error message
+      document.removeEventListener('click', handleOutsideClick);
     }
-  }, []);
+
+    return () => {
+      document.removeEventListener('click', handleOutsideClick);
+    };
+  }, [showPicker, handleOutsideClick]);
+
+  const checkIfCustomColor = (color: any) => {
+    if (!Object.values(palleteColors).includes(color)) {
+      setCustomInternalColors(customInternalColors.concat(color));
```

Comment:
What's the experience here? Are we tracking all previous attempted custom colors?

Reply:
Experience is we provide basic colors that will be displayed by default to all. But of they chose from the StechPicker and that hex code is part of the basic default color then we don't want to add those colors to part of the custom colors. This requirement is from UX. So this condition will prevent adding redundant colors to custom color list.